### PR TITLE
feature: select all records from current index pages

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Now the user is able to select all the records through all pages and run an action on all of them

Fixes https://github.com/avo-hq/avo/issues/1147

CleanShot 2022-08-23 at 13 09 57

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/controllers/avo/actions_controller.rb [hunk 1]
```diff
@@ -12,9 +12,15 @@ def show
 
     def handle
       resource_ids = action_params[:fields][:resource_ids].split(",")
-      models = @resource.class.find_scope.find resource_ids
+      selected_all = action_params[:fields][:selected_all].to_i == 1
 
-      fields = action_params[:fields].except("resource_ids")
+      models = if selected_all
+        ActiveRecord::Base.connection.exec_query base64_decode_query
```
Comment (adrianthedev) [inline app/controllers/avo/actions_controller.rb:18]: We gotta see if this is a security issue.

Comment (adrianthedev) [reply app/controllers/avo/actions_controller.rb:18]: Maybe it's better to encrypt/decrypt it with a secret key?

File: app/javascript/js/controllers/action_controller.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { Controller } from '@hotwired/stimulus'
 import { castBoolean } from '../helpers/cast_boolean'
 
 export default class extends Controller {
-  static targets = ['controllerDiv', 'resourceIds', 'form']
+  static targets = ['controllerDiv', 'resourceIds', 'form', 'selectedQuery', 'selectedAll']
```
Comment (adrianthedev) [inline app/javascript/js/controllers/action_controller.js:5]: Let's make the name a bit more obvious.

```suggestion
  static targets = ['controllerDiv', 'resourceIds', 'form', 'selectedAllQuery', 'selectedAll']
```

File: app/javascript/js/controllers/action_controller.js [hunk 3]
```diff
@@ -24,9 +26,13 @@ export default class extends Controller {
...
   get resourceIds() {
     try {
-      return JSON.parse(document.querySelector(`[data-selected-resources-name="${this.resourceName}"]`).dataset.selectedResources)
+      return JSON.parse(this.dataSet.selectedResources)
     } catch (error) {
       return []
     }
   }
+
+  get dataSet() {
```
Comment (adrianthedev) [inline app/javascript/js/controllers/action_controller.js:35]: This is a good refactor but let's name it a bit better please. How about `selectionOptions`?

File: app/javascript/js/controllers/action_controller.js (hunk 2 above)
Comment (Paul-Bob) [reply app/javascript/js/controllers/action_controller.js:5]: sure, fixing this on next commit

File: app/javascript/js/controllers/action_controller.js (hunk 3 above)
Comment (Paul-Bob) [reply app/javascript/js/controllers/action_controller.js:38]: changed and it looks better this way

File: app/controllers/avo/actions_controller.rb (hunk 1 above)
Comment (Paul-Bob) [reply app/controllers/avo/actions_controller.rb:18]: ActiveSupport::MessageEncryptor implemented

**Review by adrianthedev — APPROVED** [review]
This looks good. Just a few nitpicks

Let's wait for the designer to help with better styling the feature and then we'll merge it.

File: lib/avo/services/application_service.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module Avo
+  module Services
+    class ApplicationService
+      def self.encrypt(message:, purpose:)
+        new(message: message, purpose: purpose).encrypt
+      end
+
+      def self.decrypt(message:, purpose:)
+        new(message: message, purpose: purpose).decrypt
+      end
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply lib/avo/services/application_service.rb:10]: I think the advantage is we can call `Service.method(args)` instead `Service.new(args).method` improves readability in my opinion

File: spec/system/avo/select_all_spec.rb [hunk 5]
```diff
@@ -0,0 +1,161 @@
+require "rails_helper"
+
+RSpec.describe "SelectAll", type: :system do
+  let!(:per_page) { Avo.configuration.per_page }
+  let!(:total_fish) { 300 }
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply spec/system/avo/select_all_spec.rb:5]: sure!

File: app/javascript/js/controllers/action_controller.js [hunk 6]
```diff
@@ -2,11 +2,16 @@ import { Controller } from '@hotwired/stimulus'
 import { castBoolean } from '../helpers/cast_boolean'
 
 export default class extends Controller {
-  static targets = ['controllerDiv', 'resourceIds', 'form']
+  static targets = ['controllerDiv', 'resourceIds', 'form', 'selectedAllQuery', 'selectedAll']
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply app/javascript/js/controllers/action_controller.js:5]: true

**Pull request merged**